src/main/java/com/google/sps/servlets/PotentialMatchesServlet.java

```diff
@@ -60,28 +70,69 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
   /**
-  * Loads the next potential match for a user
+  * Loads the match information for a specified user if match information is not
+  * already in datastore.
   *
-  * <p>Currently is just loading all the possible matches and obtaining one of them.
+  * <p>Creates new match information entity for user if it had not already been stored.
   *
-  * <p>TODO(#21): Obtain potential match results from datastore instead of running the 
-  * potential match finding methods repeatedly.
+  * @param userID The ID of the user who's match information is being checked/loaded into datastore
+  */
+  private void loadUserMatchInformation(String userID) {
```

Comment:
Since the effect of this method is initializing some data if it does not yet exist, and it doesn't return anything itself, I would either:

1) Change this method to return something meaningful, e.g. the `List<String>` of potential matches, or just the `Entity` itself
2) OR, rename this method to something more descriptive of its intent, e.g. `ensureUserHasMatchInformation`

Returning the Entity or matches directly (option 1) seems more useful, since you only need to load it once in the case where it already exists, and don't need to load it at all in the case where it's newly created.

Reply:
Got it! I went with the first option and changed this method to return an Entity.